### PR TITLE
stu: init at 0.4.2

### PR DESCRIPTION
**Pull request by Nebucatnetzer**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by patka-123 — APPROVED** [review]

File: pkgs/by-name/st/stu/package.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    repo = "stu";
+    rev = "v${version}";
+    hash = "sha256-T5b3aCepUj8COrKReEaK4JeUbR7Sv7022xSCW8k8Iow=";
+  };
+
+  cargoSha256 = "sha256-DFG/9bnckqLezbitceLtM3CSnKAcQcZlv39VfbkyM/w=";
+
+  passthru.tests.version = testers.testVersion { package = stu; };
+
+  meta = with lib; {
```
Comment (isabelroses) [inline pkgs/by-name/st/stu/package.nix:24]: `with lib;` is an anti-pattern, see here https://github.com/NixOS/nixpkgs/issues/208242.

File: pkgs/by-name/st/stu/package.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  stu,
+  testers,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (isabelroses) [inline pkgs/by-name/st/stu/package.nix:9]: Consider removing the `rec` for the `let in` synatx, see here https://nix.dev/guides/best-practices.html.

**Comment by Nebucatnetzer** [comment]
> Can you update the PR title to `stu: init at 0.4.2` ?

Updated

**Review by patka-123 — APPROVED** [review]

File: pkgs/by-name/st/stu/package.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  rustPlatform,
+  stu,
+  testers,
+}:
+let
+  version = "0.4.2";
+in
+rustPlatform.buildRustPackage {
+  pname = "stu";
+  version = version;
```
Comment (isabelroses) [inline pkgs/by-name/st/stu/package.nix:13]:
```suggestion
  inherit version;
```

**Review by isabelroses — APPROVED** [review]

**Review by nicoonoclaste — APPROVED** [review]
Looks good otherwise, I'll happily merge this once you decide whether to apply the suggestion.

File: pkgs/by-name/st/stu/package.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  rustPlatform,
+  stu,
+  testers,
+}:
+let
+  version = "0.4.2";
+in
+rustPlatform.buildRustPackage {
+  pname = "stu";
+  inherit version;
```
[the remark this answers is not in the document]
Comment (isabelroses) [reply pkgs/by-name/st/stu/package.nix:13]: I suggested the `let in` following https://nix.dev/guides/best-practices.html, and having this practise suggested to me by anderson torres.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**